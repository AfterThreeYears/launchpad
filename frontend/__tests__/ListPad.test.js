/* @flow */

import React from 'react';
import { MemoryRouter, Switch } from 'react-router-dom';
import ListPad from '../ListPad';
import renderer from 'react-test-renderer';

test('renders normal', () => {
  const component = renderer.create(
    <MemoryRouter>
      <div>
        <ListPad
          pad={{
            id: 'fakeid1',
            title: 'Fake 1',
            description: 'Description 1',
            code: 'code',
            deployedCode: 'deployed code',
            url: 'http://example.com',
            user: {
              id: 'test-ids',
              githubUsername: 'example',
              pads: [],
            },
            context: [
              {
                key: 'secret',
                value: 'secret',
              },
            ],
            dependencies: [
              {
                name: 'graphql-js',
                version: '0.8.9',
              },
            ],
            defaultQuery: '{ hello }',
            defaultVariables: null,
            token: 'test-token-2',
            draft: null,
          }}
        />,
      </div>
    </MemoryRouter>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
