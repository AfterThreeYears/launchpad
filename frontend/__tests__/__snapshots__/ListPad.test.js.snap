// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders normal 1`] = `
<div>
  <div
    className="ListPad"
  >
    <a
      className="ListPad-Link"
      href="fakeid1"
      onClick={[Function]}
    >
      Pads you created
      <div
        className="ListPad-Header"
      >
        <div
          className="ListPad-HeaderUsername"
        >
          example
        </div>
        <div
          className="ListPad-HeaderTitle"
        >
          Fake 1
        </div>
      </div>
    </a>
  </div>
  ,
</div>
`;
